pkg/detectors/falsepositives.go

```diff
@@ -151,22 +152,27 @@ func FilterResultsWithEntropy(ctx context.Context, results []Result, entropy flo
 }
 
 // FilterKnownFalsePositives filters out known false positives from the results.
-func FilterKnownFalsePositives(ctx context.Context, detector Detector, results []Result, shouldLog bool) []Result {
+func FilterKnownFalsePositives(ctx context.Context, detector Detector, results []Result, retainFalsePositives bool) []Result {
 	var filteredResults []Result
 
 	isFalsePositive := GetFalsePositiveCheck(detector)
 
 	for _, result := range results {
-		if !result.Verified && result.Raw != nil {
-			isFp, reason := isFalsePositive(result)
-			if !isFp {
-				filteredResults = append(filteredResults, result)
-			} else if shouldLog {
-				ctx.Logger().Info("Filtered out known false positive", "result", result, "reason", reason)
-			}
-		} else {
+		if len(result.Raw) == 0 {
+			ctx.Logger().Error(fmt.Errorf("empty raw"), "invalid result; skipping")
+			continue
+		}
+
+		if result.Verified {
+			filteredResults = append(filteredResults, result)
+			continue
+		}
+
+		isFp, _ := isFalsePositive(result)
+		if !isFp || retainFalsePositives {
```

Comment:
Doesn't this mean that calling `FilterKnownFalsePositives` with `retainFalsePositives == true` is equivalent to not calling at all? Wouldn't that be a simpler way to handle this? (Other than the zero-length `Raw` check, but that's not a load-bearig part of this function.)

Reply:
Updated, let me know if it looks correct now.